pkg/event/event.go

```diff
@@ -0,0 +1,130 @@
+package event
+
+import (
+	"github.com/aws/aws-sdk-go/service/sns/snsiface"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+)
+
+// Publisher interface defines anything that can publish an event
+type Publisher interface {
+	Publish(i interface{}) error
+}
+
+// NewEventerInput are the items required to create a new Eventer service
+type NewEventerInput struct {
+	SnsClient              snsiface.SNSAPI
+	SqsClient              sqsiface.SQSAPI
+	AccountCreatedTopicArn string `env:"ACCOUNT_CREATED_TOPIC_ARN" defaultEnv:"DefaultAccountCreatedTopicArn"`
+	AccountDeletedTopicArn string `env:"ACCOUNT_DELETED_TOPIC_ARN" defaultEnv:"DefaultAccountDeletedTopicArn"`
+	AccountResetQueueURL   string `env:"RESET_SQS_URL" defaultEnv:"DefaultResetSQSUrl"`
+	LeaseAddedTopicArn     string `env:"LEASE_ADDED_TOPIC" defaultEnv:"DefaultLeaseAddedTopicArn"`
+}
+
+// Eventer is the public interface for publishing events
+type Eventer struct {
```

Comment:
Maybe another name? Just thinking that `er` suffixed objects are usually interfaces, by convention

Reply:
`EventHub`?

Reply:
or just `Events`?

Reply:
Switching this to Hub.  EventHub replicates Event from the package.